.github/workflows/java-cd.yml

```diff
@@ -149,12 +127,41 @@ jobs:
                   jar -cvf bundle.jar *
                   ls -ltr
                   cd -
-                  cp $src_folder/bundle.jar bundle-${{ matrix.host.TARGET }}.jar
+                  cp $src_folder/bundle.jar bundle-${{ matrix.build.TARGET }}.jar
 
             - name: Upload artifacts to publish
               continue-on-error: true
               uses: actions/upload-artifact@v4
               with:
-                  name: java-${{ matrix.host.TARGET }}
+                  name: java-${{ matrix.build.TARGET }}
                   path: |
                       java/bundle*.jar
+
+    combine-binaries-to-one-directory:
+        needs: create-binaries-to-publish
+        runs-on: ubuntu-latest
+        env:
+            INPUT_VERSION: ${{ github.event.inputs.version }}
```

Comment:
You have no input if CD started on a PR.
I think you need to move code on lines 72-86 to this job to determine the version here.